compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -853,16 +849,19 @@ public void visit(BLangMatchClause matchClause) {
 
             this.isJSONContext = types.isJSONContext(matchExprType);
             analyzeNode(matchPattern, env);
-            matchPattern.isLastPattern = this.hasLastPatternInClause;
+            this.hasLastPatternInClause = this.hasLastPatternInClause || matchPattern.isLastPattern;
         }
 
         if (!patternListContainsSameVars) {
             dlog.error(matchClause.pos, DiagnosticErrorCode.MATCH_PATTERNS_SHOULD_CONTAIN_SAME_SET_OF_VARIABLES);
         }
         matchClause.declaredVars.putAll(matchClause.matchPatterns.get(0).declaredVars);
 
+        boolean hasLastPatternInStatement = this.hasLastPatternInStatement;
```

Comment:
We can do this when visiting `BLangMatchStatement`, before L803, right? Or we will be doing this assignment and re-assignment in L864 per clause even if there is no match statement in the clause?

Reply:
Yes, changed